### PR TITLE
Api server

### PR DESCRIPTION
**Pull request by MDunitz**
# Need
* Setup API Gateway
* HealthCheck endpoint

# Approach
* Created in terraform 
* Swagger used to document endpoints (https://query.dev.data.humancellatlas.org/) 

# Testing
* Unit test
* Currently deployed in dev (https://query.dev.data.humancellatlas.org/v1/health should return a 200)

Fixes https://github.com/HumanCellAtlas/query-service/issues/31
Connects to https://github.com/HumanCellAtlas/query-service/issues/15

### CODE REVIEW

File: query/lib/common/logging.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import sys
+
+
+def get_logger(name):
+    ch = logging.StreamHandler(sys.stdout)
+    log_level_name = os.environ['LOG_LEVEL'] if 'LOG_LEVEL' in os.environ else 'DEBUG'
+    log_level = getattr(logging, log_level_name.upper())
+    ch.setLevel(log_level)
+    formatter = logging.Formatter('%(asctime)s %(levelname)s %(name)s %(message)s',
+                                  datefmt="%Y-%m-%dT%H:%M:%S%z")
```
Comment (mweiden) [inline query/lib/common/logging.py]: nit: could you make sure that this is UTC and add a `Z` at the end of the datefmt?

File: query/test/unit/api_server/test_health.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+from query.test.unit import QueryTestCaseUsingMockAWS
+from query.test.unit.api_server import client_for_test_api_server
+
+
+class TestHealthCheckEndpoint(QueryTestCaseUsingMockAWS):
+
+    def test_heathcheck_endpoint(self):
+        import pdb
+        # pdb.set_trace()
```
Comment (mweiden) [inline query/test/unit/api_server/test_health.py]: Remove this if not used.

File: query/test/unit/__init__.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            os.environ[k] = v
+
+    def __enter__(self):
+        self.enter()
+
+    def __exit__(self, type, value, traceback):
+        self.exit()
+
+
+class QueryTestCaseUsingLiveAWS(unittest.TestCase):
```
Comment (mweiden) [inline query/test/unit/__init__.py]: Not used. Try removing this.

File: query/test/unit/__init__.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    def setUp(self):
+        # Does nothing but provide for consistency in test subclasses.
+        pass
+
+    def tearDown(self):
+        # Does nothing but provide for consistency in test subclasses.
+        pass
+
+
+class QueryTestCaseUsingMockAWS(unittest.TestCase):
```
Comment (mweiden) [inline query/test/unit/__init__.py]: May not be needed. Try removing this.

File: query/test/unit/api_server/test_health.py (hunk 2 above)
Comment (MDunitz) [reply query/test/unit/api_server/test_health.py]: 👍

File: query/test/unit/__init__.py (hunk 3 above)
Comment (MDunitz) [reply query/test/unit/__init__.py]: 👍

File: query/test/unit/__init__.py (hunk 4 above)
Comment (MDunitz) [reply query/test/unit/__init__.py]: 👍

File: config/iam-policy-templates/query-api-lambda.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Resource": "arn:aws:logs:*:*:*"
```
[the remark this answers is not in the document]
Comment (MDunitz) [reply config/iam-policy-templates/query-api-lambda.json]: 👍

File: query/lib/common/logging.py (hunk 1 above)
Comment (MDunitz) [reply query/lib/common/logging.py]: 👍

**Pull request merged**